src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,131 @@
+            }
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM book WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can not connect to database", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM book";
+        List<Book> books = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement statement = connection.createStatement()) {
+
+            ResultSet resultSet = statement.executeQuery(sql);
+
+            while (resultSet.next()) {
+                Book book = new Book();
+                book.setId(resultSet.getLong("id"));
+                book.setTitle(resultSet.getString("title"));
+                book.setPrice(resultSet.getBigDecimal("price"));
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error getting books from database.", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE book SET title = ?, price = ? WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("At least one row was expected to be deleted");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error updating books in database.", e);
+        }
+        return null;
```

Comment:
The `update` method should return the updated `Book` object instead of null. Modify the return statement to return the `book` object after the update operation.